zappa/core.py

```diff
@@ -767,7 +769,9 @@ def splitpath(path):
 
         # Cleanup
         for glob_path in exclude_glob:
-            for path in glob.glob(os.path.join(temp_project_path, glob_path)):
+            # Use `recursive` to match paths deep in the directory tree
+            # https://github.com/zappa/Zappa/issues/1269
+            for path in glob.glob(os.path.join(temp_project_path, glob_path), recursive=True):
```

Comment:
Thanks for the very thoughtful and thorough review (especially for such an apparently minor change).  I'll be happy to add some tests when I have time to swing back to this.  You've asked some interesting questions about performance management and I agree it would be nice to characterize the failure modes a little more and consider how to handle or document them.  I'm not sure about including those in the test suite though--seems like intentionally triggering performance issues in a unit test will be a pain for anyone running tests going forward, though I'm open to other opinions.  

I'll have to think a little bit about what mitigations might look like in this scenario; to a degree, I feel that either preemptively bombing the process (e.g. a timeout) or executing a halfway correct solution (e.g. limiting the number of results processed) could lead to more frustrating bugs than an obvious failure with a long-running process.  The other good thing is that this affects a relatively early part of the deploy/update process so if it does hang, there isn't a lot of garbage that will be left other than the temp file tree that was being operated on.